build/Band_Representations.jl

```diff
@@ -0,0 +1,260 @@
+        P = pointgroup(iuclab, D)
+        ctᴾ = MultTable(P).table
+        if ctᴳ == ctᴾ # bit sloppy; would miss ismorphisms that are "concealed" by row/column swaps
+            return P
+        end
+    end
+    return nothing # in case we didn't find any isomorphic parent
+end
+
+"This function gives the site_symmetry irreps for the Site Symmetry Group"
+function site_symmetry_irreps(sitegroup::SiteGroup{D}, timereversal::Bool) where D
+    site_parent_pointgroup=_find_isomorphic_parent_pointgroup(sitegroup)
+
+    timereversal==false ? get_pgirreps(site_parent_pointgroup.label, Val(D)) : realify(get_pgirreps(site_parent_pointgroup.label, Val(D)))
+end
+
+"The function below gives the character table for the site symmetry group provided"
+function site_symmetry_irreps_characters(sitegroup::SiteGroup{D}, timereversal::Bool) where D
+    irreps=site_symmetry_irreps(sitegroup, timereversal)
+    character_table=Matrix{Complex{Float64}}(undef, size(irreps[1].matrices)[1], size(irreps)[1] )
+    size(character_table)
+    for j in 1:size(irreps)[1]
+        for i in 1:size(irreps[1].matrices)[1]
+            character_table[i, j]=tr(irreps[j].matrices[i]) 
+        end
+    end
+character_table
+end
+
+
+"This function provides the induced band representation"
+function induced_band_representation(sitegroup::SiteGroup{D}, h::SymOperation{D}, ρ::Int64, k::KVec, timereversal::Bool) where D
+    orbit_of_sitegroup=orbit(sitegroup, sitegroup.wp);
+    length_of_orbit=length(orbit_of_sitegroup)
+    #Now we sum over all the wyckoff positions in the orbit  
+    gα=cosets(sitegroup)
+    χ_G_k_G=0;
+    wp=sitegroup.wp;
+    wp_rvec=wp.qv.cnst
+    gα_inverse=gα[1]
+
+    transformed_k=transform(k, rotation(h)).cnst
+    irrep_matrices=site_symmetry_irreps(sitegroup, timereversal)[ρ].matrices
+    for α′ in 1:length_of_orbit 
+
+        gα′=gα[α′]
+        tα′α′=rotation(h)*orbit_of_sitegroup[α′].qv.cnst+translation(h)-orbit_of_sitegroup[α′].qv.cnst
+        symop_t = SymOperation(-tα′α′)
+        gα_inverse=inv(gα′)    
+
+        gα′⁻¹ggα′ = compose(gα_inverse, compose(symop_t, compose(h,  gα[α′], false), false ), false)
+        site_symmetry_index = findfirst(≈(gα′⁻¹ggα′), sitegroup)
+
+        if site_symmetry_index !== nothing
+            χ_G_k_G += cis(2π*dot(transformed_k, tα′α′))*tr(irrep_matrices[site_symmetry_index])
+        end
+    end
+    return χ_G_k_G
+end
+
+function subduce_onto_littlegroups(k::String, kvec::KVec, ρ::Int64, sitegroup::SiteGroup{D}, timereversal::Bool) where D
+    sgnum=num(sitegroup)
+    sg=spacegroup(sgnum, D)
+    lgirs=get_lgirreps(sgnum, D, timereversal)
+    num_lgirreps= size(lgirs[k])[1]
+    
+    size_lgirreps=size(lgirs[k][1].matrices)[1]
+
+    character_table=Matrix{Complex{Float64}}(undef, size_lgirreps, num_lgirreps )
+
+    for j in 1:num_lgirreps
+        for i in 1:size_lgirreps
+            character_table[i, j]=tr(get_lgirreps(sgnum, D, timereversal)[k][j].matrices[i])
+        end
+    end
```

Comment:
```suggestion
    character_table = CharacterTable(get_lgirreps(sgnum, D, timereversal)).chartable
```

Reply:
get_lgirreps returns a dictionary- how does one define a CharacterTable with little group irreps?

Reply:
Ah, there's a `[k]` indexing missing in my suggestion.

Reply:
See updated suggestion.